src/backend/distributed/executor/executor.c

```diff
@@ -572,7 +642,24 @@ DistributedStatementRequiresRollback(DistributedPlan *distributedPlan)
 	task = (Task *) linitial(taskList);
 	if (list_length(task->taskPlacementList) > 1)
```

Comment:
I think it is possible for a relation to have shards with non-equal replication factors, e.g. if we create shards with replication factor 2, then decomission a node, then some shards will have replication factor 2 and some other replication factor 1. In that, is checking the number of placements of one task enough?

Reply:
There is a function called `SingleReplicatedTable()`, which I can use here. However, I'm not sure whether it is worth using that.

Because, at this point we're already passed `list_length(taskList) > 1`, in the sense that there is only a single task we're dealing with for the distributed plan. And, (a) We'll only return true for `reference tables` after this point, otherwise always false (b) `SingleReplicatedTable()` or any other such function has to iterate through all the shards/placements to decide, which might be too large for append partitioned tables.

I'm more inclined to keep this as is and maybe add a comment. What do you think @pykello